tests/ui/auto-traits/issue-83857-ub.rs

```diff
@@ -0,0 +1,31 @@
+#![allow(suspicious_auto_trait_impls)]
+
+struct Always<T, U>(T, U);
+unsafe impl<T, U> Send for Always<T, U> {}
+struct Foo<T, U>(Always<T, U>);
+
+trait False {}
+unsafe impl<U: False> Send for Foo<u32, U> {}
```

Comment:
This would be worth a comment explaining what is being tested: that the manual `impl Send for Foo` should "block" the implicit impl